### ISSUE TITLE
New API Search Delays Startup and Interrupts Script - Possible Improvements

### ISSUE THREAD

**Issue by rpseleven** [beaston02/SMrecorder#7]
Hello,

New to GitHub and not a native English speaker, so please excuse any odd behavior. I appreciate your work on this script.

I have noticed the newest version starts and immediately searches to see if models are in some way "blocked" and pushes their recording through a different syntax. My problem with this is just how long the requests take to complete to populate the nonApiModels variable when there are 50+ requests to make. There may be many minutes between the script starting and actual recording happening, and this is with a little-used 150 Mbps service. This is more a problem when you schedule the start and stop of the script to either split up recordings, or ensure the script is actually running 24/7.

I was thinking about exporting the nonApiModels to a file and importing it on start, then starting recording, but then you still need to run findNonApiModels() after, which interrupts the periodic getOnlineModels(). Not ideal.

The only other solution I can think of is running a separate script or thread to maintain a nonApiModels list separately from the main loop, and just import the nonApiModels file every interval check. I have programmed it just to call a stub of a script containing findNonApiModels() which dumps to a text file nonApiModels every time the while loop exits. Ugly, but functional.

I'm sure beaston02 or another user can come up with a vastly superior solution.

Also, I'm not entirely sure on this, as it could just be my operating environment or Python version, but in testing shorter intervals between exiting the main while loop, I could not get the loop to exit after 4 hours. I have a feeling it has to do with a misplaced multiplier and variable on line 143. It is odd as the variable "checked" is instantiated, but never used.

Lines 142 and 143 of the script:

```
 checked = int(time.time())
 while int(time.time()) < int(time.time()) * 60 * 60 * 4:
```
What I think it should be?:

```
 checked = int(time.time())
 while int(time.time()) < checked + (60 * 60 * 4):
```

Example: This exits after 3 minutes:

```
 checked = int(time.time())
 while int(time.time()) < checked + (60 * 3):
```

Thank you for your help.

**Comment by beaston02** [comment]
Thank you for pointing this out. What I did was created a thread to run the check for non api models so it doesnt stop the rest of the script from running, but that way will also check periodically in the event a model changes to not be listed on the api (I believe these are models who you must be logged in to see them in the directory, or anywhere other than going straight to their profile - another user pointed this out to me, otherwise I would have never known this was the case). I did not add a file to add the models to yet. I would kind of like this to all be handled automatically, so I decided to try it this way first. 

The code has all been updated. let me know if you have any issues! 

Thanks again!
